### PR TITLE
feat: disable closing plugin drawer on clicking the mask

### LINKED ISSUE #1548: Clicking on a mask while editing a plugin does not allow the drawer to be closed
# Feature request

Users may click on the mask by mistake when editing complex plugins, and the drawer will close after clicking on the mask, resulting in the loss of data edited by the user.
image

## Describe the solution you'd like
Clicking on the mask does not allow the drawer to be closed

We can set `maskClosable ` to `false` to do this. 

Refer link:  https://ant.design/components/drawer/#API

cc @tokers

### PR DESCRIPTION
**Pull request by bisakhmondal**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
resolves #1548 

### New feature or improvement
Now, clicking on the masked area of the plugin drawer does not close it.

### CODE REVIEW

**Comment by LiteSun** [comment]
Very glad to see you pr! The drawer in route and service also needs to be handled.
You can search globally for this component to prevent missing some pages
 😁 @bisakhmondal 

image

**Comment by bisakhmondal** [comment]
>Very glad to see you pr! The drawer in route and service also needs to be handled.

Yeah sure. On issue #1548, you only mentioned the plugins, that's why I left the rest of them as it is.
Pushing the changes soon.
Thanks for the review @LiteSun :)

File: web/src/components/Plugin/PluginDetail.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ type Props = {
   pluginList: PluginComponent.Meta[];
   readonly?: boolean;
   visible: boolean;
+  maskClosable: boolean;
```
Comment (juzhiyuan) [inline web/src/components/Plugin/PluginDetail.tsx:49]:
```suggestion
  maskClosable?: boolean;
```

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Comment by bisakhmondal** [comment]
>The drawer in route and service also needs to be handled.
You can search globally for this component to prevent missing some pages

Hey @LiteSun, for the plugins @ route and service, components are reflecting the current changes. It's using the existing `PluginPage`  component (on which the changes have been made) for rendering the plugins in different use cases.

for example inside `service`
https://github.com/apache/apisix-dashboard/blob/6dce9e09640b1c81f14dbb88e5f6f8c43371af31/web/src/pages/Service/Create.tsx#L133-L135

in `routes`
https://github.com/apache/apisix-dashboard/blob/6dce9e09640b1c81f14dbb88e5f6f8c43371af31/web/src/pages/Route/components/Step3/index.tsx#L86-L95

In short, if I am not mistaken, desired changes are getting reflected for plugins across the components you mentioned.

Do you want me to add an extra optional prop (`maskClosable`) for `PluginPage` itself to get a more precise control over the mask to use it from any page?

Thanks for the approval @juzhiyuan 😄

**Comment by bisakhmondal** [comment]
Oops! one test is failing. But at my local machine all tests of `test:e2e` are working fine.

I have made available the full test report at [pastebin](https://pastebin.com/pF7P02U9). 
Could you please look into it? Thank you.


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  consumer/create_and_delete_consumer      00:13        4        4        -        -        - │
  │    .spec.js                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  lang/switch-language.spec.js             00:05        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  metrics/metrics-smoketest.spec.js        00:02        1        1        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  plugin/create-edit-delete-plugin.sp      16:43        3        3        -        -        - │
  │    ec.js                                                                                       │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  pluginTemplate/create-edit-delete-p      00:12        3        3        -        -        - │
  │    lugin-template.spec.js                                                                      │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  pluginTemplate/create-plugin-templa      00:20        4        4        -        -        - │
  │    te-with-route.spec.js                                                                       │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/can-skip-upstream-when-select      00:20        4        4        -        -        - │
  │    -service-id.spec.js                                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/create-edit-delete-route.spec      00:22        4        4        -        -        - │
  │    .js                                                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/create-route-with-upstream.sp      00:19        4        4        -        -        - │
  │    ec.js                                                                                       │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/import_export_route.spec.js        00:25        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/online-debug.spec.js               00:29        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  route/search-route.spec.js               00:25        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  service/create-edit-delete-service.      00:16        4        4        -        -        - │
  │    spec.js                                                                                     │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  service/edit-service-with-upstream.      00:18        4        4        -        -        - │
  │    spec.js                                                                                     │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  settings/settings-smoketest.spec.js      00:10        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  ssl/ssl-smoketest.spec.js                01:09        3        3        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  upstream/create_and_delete_upstream      00:19        6        6        -        -        - │
  │    .spec.js                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user/login.spec.js                       00:05        4        4        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  user/logout.spec.js                      00:04        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        22:23       64       64        -        -        -

**Comment by LiteSun** [comment]
> Do you want me to add an extra optional prop (`maskClosable`) for `PluginPage` itself to get a more precise control over the mask to use it from any page?
> 
> Thanks for the approval @juzhiyuan 😄

The current code is ok.
